embedded/src/main/java/org/wildfly/core/embedded/EmbeddedServerFactory.java

```diff
@@ -338,4 +340,44 @@ private static void setupLoggingSystem(ModuleLoader moduleLoader) {
             WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(tccl);
         }
     }
+
+    private static Object createServer(final Method createServerMethod, final ModuleLoader moduleLoader, final File jbossHomeDir, final String[] cmdargs) {
+        Object serverImpl;
+        try {
+            Properties sysprops = WildFlySecurityManager.getSystemPropertiesPrivileged();
+            Map<String, String> sysenv = WildFlySecurityManager.getSystemEnvironmentPrivileged();
+            String[] args = cmdargs != null ? cmdargs : new String[0];
+            serverImpl = createServerMethod.invoke(null, jbossHomeDir, moduleLoader, sysprops, sysenv, args);
+        } catch (final InvocationTargetException ite) {
+            throw EmbeddedLogger.ROOT_LOGGER.cannotCreateStandaloneServer(ite.getCause(), createServerMethod);
+        } catch (final IllegalAccessException iae) {
+            throw EmbeddedLogger.ROOT_LOGGER.cannotCreateStandaloneServer(iae, createServerMethod);
+        }
+        return serverImpl;
+    }
+
+    private static void resetEmbeddedServerProperties(final String jbossHomeDir, final EmbeddedServerType embeddedServerType) {
+        assert jbossHomeDir != null;
+
+        String jbossBaseDir;
+        switch (embeddedServerType) {
+            case STANDALONE:
+                jbossBaseDir = jbossHomeDir + File.separator + "standalone";
+                break;
+            case HOST_CONTROLLER:
+                jbossBaseDir = jbossHomeDir + File.separator + "domain";
+                break;
+            default:
+                throw new RuntimeException("Unknown embedded server type: " + embeddedServerType);
+        }
+
+        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_HOME_DIR, jbossHomeDir);
+        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_BASE_DIR, jbossBaseDir);
+        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_CONFIG_DIR, jbossBaseDir + File.separator + "configuration");
+        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_DEPLOY_DIR, jbossBaseDir + File.separator + "data" + File.separator +"content");
+        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_TEMP_DIR, jbossBaseDir + File.separator + "data" + File.separator + "tmp");
+        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_LOG_DIR, jbossBaseDir + File.separator + "log");
+        WildFlySecurityManager.setPropertyPrivileged(SYSPROP_KEY_JBOSS_DATA_DIR, jbossBaseDir + File.separator + "data");
```

Comment:
The jboss.server.xxx.dir properties should not be being set if the embedded process is a Host Controller. The HC equivalents are jboss.domain.xxx.dir.